app/models/access_control.rb

```diff
@@ -1,8 +1,17 @@
 # frozen_string_literal: true
 
 class AccessControl < ApplicationRecord
+  class Level
+    DISCOVER = 'discover'
+    READ = 'read'
+    EDIT = 'edit'
+  end
+
   belongs_to :agent,
              polymorphic: true
   belongs_to :resource,
              polymorphic: true
+
+  validates :access_level,
+            uniqueness: { scope: [:agent, :resource] }
```

Comment:
nitpick: we should also probably validate that `access_level` is one of our Level constants, similar to how we do `Work#work_type`

Reply:
@rschenk yes, good catch! I will add that in...